### PR TITLE
dm(engine): Discard resource when it's no longer needed

### LINKED ISSUE #6784: DMWorkerSync should not request resource after metadata is loaded into checkpoint
### Is your feature request related to a problem?

to provide more freedom for scheduling

### Describe the feature you'd like

as title

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #6784

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/executor/dm/worker.go [hunk 1]
```diff
@@ -318,3 +330,27 @@ func (w *dmWorker) checkAndAutoResume(ctx context.Context) error {
...
+
+func (w *dmWorker) discardResource4Syncer(ctx context.Context) {
+	if w.workerType != frameModel.WorkerDMSync || !w.needExtStorage {
+		return
+	}
+	impl, ok := w.unitHolder.(*unitHolderImpl)
+	if !ok {
+		return
+	}
+	isFresh, err := impl.unit.IsFreshTask(ctx)
```
Comment (GMHDBJD) [inline engine/executor/dm/worker.go:342]: So we need to judge in every Tick?

Comment (lance6716) [reply engine/executor/dm/worker.go:342]: yes, until we set `needExtStorage` to `false`

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**